tests/unit/execution_tree/primitives/store_operation.cpp

```diff
@@ -9,38 +10,362 @@
 #include <hpx/include/lcos.hpp>
 #include <hpx/util/lightweight_test.hpp>
 
+#include <string>
 #include <utility>
 #include <vector>
 
+///////////////////////////////////////////////////////////////////////////////
+phylanx::execution_tree::primitive_argument_type compile_and_run(
+    std::string const& codestr)
+{
+    phylanx::execution_tree::compiler::function_list snippets;
+    phylanx::execution_tree::compiler::environment env =
+        phylanx::execution_tree::compiler::default_environment();
+
+    auto const& code = phylanx::execution_tree::compile(codestr, snippets, env);
+    return code.run();
+}
+/////////////////////////////////////////////////////////////////////////////
 void test_store_operation()
 {
-    phylanx::execution_tree::primitive lhs =
-        phylanx::execution_tree::primitives::create_variable(
-            hpx::find_here(), phylanx::ir::node_data<double>(0.0));
+    std::string const code = R"(block(
+        define(a,57.7),
+        store(a,42.0),
+        a
+    ))";
+
+    auto result =
+        phylanx::execution_tree::extract_numeric_value(compile_and_run(code));
+
+    HPX_TEST_EQ(result[0], 42.0);
```

Comment:
Please add a check for `result.size() == 1` before blindly accessing `result[0]`. Same needs to happens for the for tests below.

Reply:
done.